presto-main/src/main/java/com/facebook/presto/execution/QueryTracker.java

```diff
@@ -91,6 +94,7 @@ public QueryTracker(QueryManagerConfig queryManagerConfig, ScheduledExecutorServ
         this.maxQueryRunningTaskCount = queryManagerConfig.getMaxQueryRunningTaskCount();
 
         this.queryManagementExecutor = requireNonNull(queryManagementExecutor, "queryManagementExecutor is null");
+        this.queryTrackerService = queryTrackerService;
```

Comment:
can `queryTrackerService` be null / does guice guarantee an `Optional.empty` if there is no wiring.

Reply:
Yes guice optional binding will bind it default to `Optional.empty` if binding is not present.